clang/lib/Sema/SemaCast.cpp

```diff
@@ -2092,6 +2092,10 @@ void Sema::CheckCompatibleReinterpretCast(QualType SrcType, QualType DestType,
     }
   }
 
+  if (SrcTy->isTemplateTypeParmType() || DestTy->isTemplateTypeParmType()) {
```

Comment:
This should probably be`if (SrcTy->isDependentType() || DestTy->isDependentType())`. Otherwise we will continue to warn for the following:
```cpp
template<typename T, typename U>
void f(T** x) {
  *reinterpret_cast<U**>(x);
}
```

Reply:
Good point
Fixed in 541e0af03998